File: MTGSearch/src/main/java/com/dbottillo/mtgsearchfree/view/adapters/OnCardListener.java
package com.dbottillo.mtgsearchfree.view.adapters;

import android.view.MenuItem;
import android.view.View;

import com.dbottillo.mtgsearchfree.model.MTGCard;

public interface OnCardListener {
    void onCardsViewTypeSelected();
    void onCardsSettingSelected();
    void onCardSelected(MTGCard card, int position, View view);
    void onOptionSelected(MenuItem menuItem, MTGCard card, int position);
}

File: MTGSearch/src/main/kotlin/com/dbottillo/mtgsearchfree/ui/cards/OnCardListener.kt
package com.dbottillo.mtgsearchfree.ui.cards

import android.view.MenuItem

import com.dbottillo.mtgsearchfree.model.MTGCard

interface OnCardListener {
    fun onTitleHeaderSelected()
    fun onCardsViewTypeSelected()
    fun onCardsSettingSelected()
    fun onCardSelected(card: MTGCard, position: Int)
    fun onOptionSelected(menuItem: MenuItem, card: MTGCard, position: Int)
}